### ISSUE TITLE
FR: Prerender pages not i view

### ISSUE THREAD

**Issue by barnewsapp** [stephanrauh/ngx-extended-pdf-viewer#950]
**Is your feature request related to a problem? Please describe.**
I think this viewer and pdf.js works well displaying our PDF-files. 
The biggest issue is that the performance when the pages is rendering. 
Our PDFs are newspaper-PDFs with somewhat complex layout with a lot of images and ads. Typically 30-80 pages. 
 
This is not an issue after the page is rendered the first time. 
If I go til page 2-3, then 4-5 and then back to 2-3 again it displays really fast. 

**Describe the solution you'd like**
Is it possible to add a feature to "prerender" the 2-3 next/previous pages in the document, so that when the user actually views the next page, the page is displays much faster? 

I know PDFJS Express (also based on pdfjs) has this feature: 
https://pdfjs.express/documentation/learn-more/events#documentviewerpagecomplete
https://pdfjs.express/api/Core.html#.SetPreRenderLevel__anchor

> A few extra pages may be prerendered at lower priority, so pageComplete may be called for pages that are not currently visible. You can set the prerender level with the SetPreRenderLevel function.

**Describe alternatives you've considered**
Maybe if would be possible to load the document in a hidden (invisible) part of the page, and load the pages in that container? 
Or maybe not, there can only be one pdf-viewer on a page. 

Thanks

**Comment by stephanrauh** [comment]
Yes, that's possible. Truth to tell, I thought pdf.js does this out-of-the-box. It's a surprise to learn it doesn't.

Can you send me one of your documents so I can reproduce the issue? If you don't want to make them publicly visible, just drop me an e-mail (webmaster1 at beyondjava dot de). 

I also wonder if it makes sense to expose the pre-rendering API, allowing users to implement their own pre-rendering strategy. For instance, you could prioritize the first page of each chapter in a book.

**Comment by barnewsapp** [comment]
Sure, I'll send you one of the PDFs to your email. 

But after checking some more it may look like it is only issue when using spread mode. 

Because the left side PDF is rendered fast, but the right one I can see is being rendered when changing page til 2-3, 4-5 and so on. 

So maybe just the next (left) page is prerendered now, the fix to also prerender the next (right) page should be easy?
